packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -668,13 +669,7 @@ abstract class TimelineEvent extends TreeNode<TimelineEvent> {
   bool get isWellFormedDeep => _isWellFormedDeep(this);
 
   bool _isWellFormedDeep(TimelineEvent event) {
-    if (!event.isWellFormed) {
-      return false;
-    }
-    for (var child in event.children) {
-      return _isWellFormedDeep(child);
-    }
-    return true;
+    return !containsChildWithCondition((e) => !e.isWellFormed);
```

Comment:
aren't you now missing verifying that the root node is well formed?

Reply:
The first node checked is the root

Reply:
"containsChildWithCondition" is a bit of a misleading name.
How about, deepWithCondition or subtreeHasNodeWithCondition.
constainsChildWithCondition doesn't make it clear it is deep or that it includes the root node as well as the children.

Reply:
renamed method to `subtreeHasNodeWithCondition`